refactor: Use selectors for core network state access

Gudahtt opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**

Replace direct references to NetworkController state with selectors. Most of these examples were skipped in previous selector refactors because they were outside of a React context, so there was no established way to get the Redux state as input for the selector. We can now do that by importing the Redux store directly and calling `getState`.

**Issue**

This relates to https://github.com/MetaMask/mobile-planning/issues/1226

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

Gudahtt committed:
refactor: Use selectors for core network state access

Replace direct references to NetworkController state with selectors.
Most of these examples were skipped in previous selector refactors
because they were outside of a React context, so there was no
established way to get the Redux state as input for the selector. We
can now do that by importing the Redux store directly and calling
`getState`.

This relates to https://github.com/MetaMask/mobile-planning/issues/1226

Gudahtt commented:
Passing E2E smoke test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/fd4f0927-b8f0-4c50-a772-50c307cc9154

Cal-L commented on app/core/WalletConnect/WalletConnect.js:
```diff
@@ -282,8 +283,9 @@ class WalletConnect {
-    const chainId =
-      Engine.context.NetworkController.state.providerConfig.chainId;
+    const chainId = selectChainId({
+      engine: { backgroundState: Engine.context },
```
Is it possible to use store.getState() here?

Gudahtt replied:
Yikes! Absolutely, sorry I missed this case

Gudahtt replied:
Fixed here: 78a137cd8808b7d94cab36ebdfd236e5e57bbf23

Cal-L reviewed (commented):
Left a comment

Gudahtt committed:
Update WalletConnect to pass full state to selector

The WalletConnect module has been updated to pass the full Redux state
to the `selectChainId` selector, rather than just the background state.
The background state was being passed in alone because it was left
behind accidentally from an earlier version of this branch, from before
the recent store refactors/TypeScript conversion.

Cal-L reviewed (approved):
LGTM
